### PR TITLE
use major,minor,build version when specifying a LibG verison 

### PR DESCRIPTION
**Pull request by mjkkirschner**
### Purpose
~~⚠️  This is a WIP~~
This PR enables specifying more specific versions of libG than those defined only by the major version number like `LibG_223` -  instead the LibG folders should have the structure `libG_major_minor_build`.

`LibraryVersionEnum` is made obsolete and instead we use `system.version` to represent the libG versions.

I have added overloads or new methods with the suffix `2` for some of these methods which accept a list of `system.versions` instead of `LibraryVersion` objects. We cannot remove these old public methods now as old clients might still call them.

I am writing tests now.

updated DynamoRevit clients will consume the method `GetGeometryFactoryPath2` to find the geometryFactoryPath using the same version that revit preloads. Please see the associated dynamoRevit PR.

If an old client calls for a library version to be loaded like `223` - it will be mapped to the real version via a map of the versions we have previously released. This means all libG folders can have the more specific naming scheme.

```c#
public static string GetGeometryFactoryPath2(string rootFolder, Version version)
```

I've also added a new configuration parameter to the testConfig file - `RequestedLibraryVersion2` - I'm not familiar with the test config - I see that it does exist in the Revit repo with ASM 223 as the specified version for 2019 - and 2020( not updated yet?) @aparajit-pratap do you know if this test config is even used during our builds/test runs? I am going to investigate that on the revit branches after I get the tests running in RTF there - I will see how libG is loaded in revit during normal use and also RTF testing.

### TODO:
- [x] Add new Tests 
- [x] DynamoCore 1.3.4 cherry pick
- [x] DynamoCore 2.02 cherry pick
- [ ] DynamoRevit 1.3.4 branch cherry picks (2017,2018,2019)
- [ ] DynamoRevit 2.0 branch cherry picks (2018,2019) (I think this is correct?)
- [x] Rename libG folder names so tests will run at all

### Declarations

Check these if you believe they are true

- [ ] The code base is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions), and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.

### Reviewers

Not ready for review
### FYIs

@aparajit-pratap 
@kronz

### CODE REVIEW

File: src/Tools/DynamoShapeManager/Utilities.cs [hunk 1]
```diff
@@ -88,6 +91,73 @@ public static LibraryVersion GetInstalledAsmVersion(List<LibraryVersion> version
             return LibraryVersion.None;
         }
 
+        public static Version GetInstalledAsmVersion2(
+            List<Version> versions,
```
Comment (aparajit-pratap) [inline src/Tools/DynamoShapeManager/Utilities.cs]: `IEnumerable` is preferred arg type for public APIs.

File: src/Tools/DynamoShapeManager/Utilities.cs [hunk 2]
```diff
@@ -88,6 +91,73 @@ public static LibraryVersion GetInstalledAsmVersion(List<LibraryVersion> version
             return LibraryVersion.None;
         }
 
+        public static Version GetInstalledAsmVersion2(
+            List<Version> versions, 
+            ref string location,
+            string rootFolder,
+            Func<string,IEnumerable> getASMInstallsFunc = null)
```
Comment (aparajit-pratap) [inline src/Tools/DynamoShapeManager/Utilities.cs]: Don't you have an auto-formatter installed in your VStudio :) ?

File: src/Tools/DynamoShapeManager/Utilities.cs [hunk 3]
```diff
@@ -143,7 +213,27 @@ public static void PreloadAsmFromPath(string preloaderLocation, string asmLocati
         /// FileNotFoundException.</param>
         /// <returns>The full path to GeometryFactoryAssembly assembly.</returns>
         /// 
+        [Obsolete("Please use methods marked with 2, and use specific versions instances to select a suported geometry library version to load")]
```
Comment (aparajit-pratap) [inline src/Tools/DynamoShapeManager/Utilities.cs]: Again, why not explicitly name the new method in this message?

File: src/DynamoCore/DynamoCore.csproj [hunk 4]
```diff
@@ -395,11 +396,14 @@ limitations under the License.
...
     <Copy SourceFiles="@(ExternProtoGeometryUICulture)" DestinationFolder="$(OutputPath)\$(UICulture)" />
     <Copy SourceFiles="@(ExternProtoGeometryLibGLocale)" DestinationFolder="$(OutputPath)\libg_locale\%(RecursiveDir)" />
-    <Copy SourceFiles="@(ExternLibG220)" DestinationFolder="$(OutputPath)libg_220\" />
-    <Copy SourceFiles="@(ExternLibG221)" DestinationFolder="$(OutputPath)libg_221\" />
-    <Copy SourceFiles="@(ExternLibG222)" DestinationFolder="$(OutputPath)libg_222\" />
-    <Copy SourceFiles="@(ExternLibG223)" DestinationFolder="$(OutputPath)libg_223\" />
-    <Copy SourceFiles="@(ExternLibG224)" DestinationFolder="$(OutputPath)libg_224\" />
+    
+    <Copy SourceFiles="@(ExternLibG220)" DestinationFolder="$(OutputPath)libg_220_0_0\" />
+    <Copy SourceFiles="@(ExternLibG221)" DestinationFolder="$(OutputPath)libg_221_0_0\" />
```
Comment (aparajit-pratap) [inline src/DynamoCore/DynamoCore.csproj]: Why don't we just get rid of 220 and 221?

Comment (mjkkirschner) [reply src/DynamoCore/DynamoCore.csproj]: briefly spoke about this - we'll keep these for now for flexibility, but don't intend on updating these versions any longer.

**Comment by mjkkirschner** [comment]
will merge this and 2.02 and push nugets.

**Pull request merged**